chemprop/models/mpn.py

```diff
@@ -204,25 +204,39 @@ def forward(self,
         :return: A PyTorch tensor of shape :code:`(num_molecules, hidden_size)` containing the encoding of each molecule.
         """
         if type(batch[0]) != BatchMolGraph:
+            # Group first molecules, second molecules, etc for mol2graph
+            batch = [[mols[i] for mols in batch] for i in range(len(batch[0]))]
```

Comment:
Isn't the first dimension of the batch the number of molecules, and the second the number of datapoints per batch? If so, why rearrange? If not so, are the checks in 217 and 227 ok? I am not very familiar with the multi-molecule code. The docstring says that the outer list contains the different molecules, the inner list the datapoints within a batch. Is that not true for smiles input and molecule input? If so, we should change the docstring (the same change in docstring would then be necessary in chemprop/models/mpn.py, too)

Reply:
@hesther you're totally right about something being wrong there. I think the documentation is wrong. The `MoleculeDataset` returns a list of lists where the outer list is the number of data points and the inner list is the number of molecules per data point, which is the flip of what the documentation had said. I just fixed the documentation. I believe the new code should still work correctly since I wrote it assuming that type of input even though the documentation had said differently.

Reply:
Thanks for the fix - the wrong documentation is my fault - the description was missing for quite some time, and someone suggested to add it together with a couple of other fixes. From the previous code (how mol2graph is called on batch) I assumed it is the other way round - but seems like that code was wrong/missing the line changing the dimensions. Thanks for fixing it!